enaml/core/import_hooks.py

```diff
@@ -368,26 +423,203 @@ def get_code(self):
-            code = self._load_cache(file_info)
-            return (code, file_info.src_path)
 
-        # Use the cached file if it exists and is current
-        src_mod_time = int(os.path.getmtime(file_info.src_path))
-        if os.path.exists(file_info.cache_path):
-            magic, ts = self._get_magic_info(file_info)
-            if magic == MAGIC and src_mod_time <= ts:
-                code = self._load_cache(file_info, set_src=True)
-                return (code, file_info.src_path)
+        # Try to load from the cache
+        cache = self.get_cached_code(file_info.src_path)
+        if cache is not None:
+            return cache
 
         # Otherwise, compile from source and attempt to cache
-        return self.compile_code()
+        return self.compile_code(file_info.src_path)
+
+
+#------------------------------------------------------------------------------
+# Enaml Zip Importer
+#------------------------------------------------------------------------------
+class EnamlZipImporter(EnamlImporter):
+
+    #: For potential future changes.
+    supported_archives = {
+        '.zip': ZipFile,
+    }
+
+    @classmethod
+    def locate_module(cls, fullname, path=None):
+        """ Searches for the given Enaml module within a zip and returns an instance
+        of this class on success.  
+        
+        Caching
+        ---------
+        If cache files exist within the archive they are used, otherwise cache files 
+        are created using the standard EnamlImporter's `_write_cache` method. Subsequent 
+        loads be imported by the standard EnamlImporter.
+        
+
+        Paramters
+        ---------
+        fullname : string
+            The fully qualified name of the module.
+
+        path : list or None
+            The subpackage __path__ for submodules and subpackages
+            or None if a top-level module.
+
+        Returns
+        -------
+        results : Instance(AbstractEnamlImporter) or None
+            If the Enaml module is located an instance of the importer
+            that will perform the rest of the operations is returned.
+            Otherwise, returns None.
+
+        """
+        # We're looking inside a package and 'path' the package path
+        if path is not None:
+            modname = fullname.rsplit('.', 1)[-1]
+            pkgpath = fullname.split('.')[:-1]
+            leaf = ''.join((modname, os.path.extsep, 'enaml'))
+            for stem in path:
+                enaml_path = os.path.join(stem, leaf)
+                file_info = make_file_info(enaml_path)
+
+                # Strip package off path to get the archive name
+                archive_path = stem
+                for p in pkgpath:
+                    archive_path = os.path.dirname(archive_path)
+
+                # To check if cache file is in zip file
+                cache_path = os.path.relpath(file_info.cache_path,
+                                             archive_path).replace("\\","/")
```

Comment:
Missing space after comma

Reply:
Also why do you convert back slashes ? From the docs (that I skimmed quickly) ZipFile can handle windows path just fine.